src/Components/NotFoundErrorPage.tsx

```diff
@@ -1,12 +1,25 @@
 import { useEffect } from 'react'
-import { connect, ContentTag, Obj } from 'scrivito'
+import {
+  connect,
+  ContentTag,
+  ensureUserIsLoggedIn,
+  isUserLoggedIn,
+  Obj,
+} from 'scrivito'
 
 // Make sure, that you have a proxy running for these URLs, otherwise you'll see an endless loop.
 const RELOAD_SUBPATHS = ['/auth']
 
 export const NotFoundErrorPage = connect(function NotFoundErrorPage() {
   const root = Obj.root()
 
+  useEffect(() => {
+    if (root) return
+    if (isUserLoggedIn()) return
+
+    ensureUserIsLoggedIn()
```

Comment:
* [ ] This logic may try to log users in on a 404 even if the homepage exists. `Obj.root()` is `null` while it's loaded from the backend.